intermediate_source/tensorboard_profiler_tutorial.py

```diff
@@ -287,28 +289,46 @@ def train(data):
+# To profile memory, ``profile_memory`` must be set to ``True`` in arguments of ``torch.profiler.profile``.
 #
-# Note: Because of the current non-optimized implementation of PyTorch profiler,
-# enabling ``profile_memory=True`` will take about several minutes to finish.
-# To save time, you can try our existing examples first by running:
+# You can try it by using existing example on Azure
 #
 # ::
 #
-#     tensorboard --logdir=https://torchtbprofiler.blob.core.windows.net/torchtbprofiler/demo/memory_demo
+#     pip install azure-storage-blob
+#     tensorboard --logdir=https://torchtbprofiler.blob.core.windows.net/torchtbprofiler/demo/memory_demo_1_10
 #
-# The profiler records all memory allocation/release events during profiling.
-# For every specific operator, the plugin aggregates all these memory events inside its life span.
+# The profiler records all memory allocation/release events and allocator's internal state during profiling.
+# The memory view consist of three components as shown in the following.
 #
 # .. image:: ../../_static/img/profiler_memory_view.png
 #    :scale: 25 %
 #
+# The components are memory curve graph, memory events table and memory statistics table, from top to bottom, respectively.
+#
 # The memory type could be selected in "Device" selection box.
-# For example, "GPU0" means the following table only shows each operator’s memory usage on GPU 0, not including CPU or other GPUs.
+# For example, "GPU0" means the following table only shows each operator's memory usage on GPU 0, not including CPU or other GPUs.
+#
+# The memory curve shows the trends of memory consumption. You can left click and drag on it to select events in the
+# desired range:
+#
+# .. image:: ../../_static/img/profiler_memory_curve_selecting.png
+#    :scale: 25 %
+#
+# After selection, the three components will be updated for the restricted time range, so that you can gain more
+# infomation about it. By repeating this process, you can zoom into very fine-grained detail. Right click on the graph
+# will reset to the initial state.
+#
+# .. image:: ../../_static/img/profiler_memory_curve_single.png
+#    :scale: 25 %
 #
-# The "Size Increase" sums up all allocation bytes and minus all the memory release bytes.
+# In the memory events table, the allocation and release events are paired into one entry. The "operator" column shows
+# the immediate ATen operator that is causing the allocation.  The "Allocation Time", "Release Time" and "Duration"
+# columns' data might be missing if the event occurs outside of the time range.
 #
-# The "Allocation Size" sums up all allocation bytes without considering the memory release.
+# In the memory statistics table, the "Size Increase" column sums up all allocation size and minus all the memory
+# release size, that is, the net increase of memory usage after this opeartor. And the "Allocation Size" column sums up
+# all allocation without considering the memory release, instead.
```

Comment:
Is it better to explain difference of "Self Size Increase" with "Size Increase"? User may be confused that there are 5 operators with same "Size Increase" value.

Reply:
I added some low level details to help the user to obtain a better understanding.